pgrx/src/spi.rs

```diff
@@ -366,7 +361,7 @@ impl Spi {
     /// This function will panic if for some reason it's unable to "connect" to Postgres' SPI
     /// system.  At the time of this writing, that's actually impossible as the underlying function
     /// ([`pg_sys::SPI_connect()`]) **always** returns a successful response.
-    pub fn connect<R, F: FnOnce(SpiClient<'_>) -> R>(f: F) -> R {
+    pub fn connect<R, F: FnOnce(&mut SpiClient) -> R>(f: F) -> R {
```

Comment:
I'm worried about contravariance, though if I remember correctly the lifetime parameter was already contravariant.

Reply:
I had a thought that this would now get us closer to how scoped threads work -- like we were discussing on discord earlier today.  

I haven't looked at this yet tho.

This formulation at least clearly indicates that SpiThings are borrowed from the mutable SpiClient instance, rather than needing to live as long as its lifetime.  At least that's how I see it in my head.